frontend/src/scenes/authentication/InviteSignup.tsx

```diff
@@ -221,49 +232,73 @@ function UnauthenticatedAcceptInvite({ invite }: { invite: PrevalidatedInvite })
+                                className="ph-ignore-input"
+                                data-attr="password"
+                                placeholder="••••••••••"
+                                autoComplete="new-password"
+                                autoFocus={window.screen.width >= 768} // do not autofocus on small-width screens
+                                disabled={isSignupSubmitting}
+                            />
+                        </LemonField>
 
-                <LemonField
-                    name="first_name"
-                    label="First Name"
-                    help={
-                        invite?.first_name ? 'Your name was provided in the invite, feel free to change it.' : undefined
-                    }
-                >
-                    <LemonInput data-attr="first_name" placeholder="Jane" />
-                </LemonField>
+                        <LemonField
+                            name="first_name"
+                            label="First Name"
+                            help={
+                                invite?.first_name
+                                    ? 'Your name was provided in the invite, feel free to change it.'
+                                    : undefined
+                            }
+                        >
+                            <LemonInput data-attr="first_name" placeholder="Jane" />
+                        </LemonField>
 
-                <SignupRoleSelect />
+                        <SignupRoleSelect />
+                    </>
+                )}
 
-                <LemonButton
-                    type="primary"
-                    status="alt"
-                    htmlType="submit"
-                    data-attr="password-signup"
-                    loading={isSignupSubmitting}
-                    center
-                    fullWidth
-                >
-                    Continue
-                </LemonButton>
+                {precheckResponse.status === 'pending' || !precheckResponse.sso_enforcement ? (
+                    <LemonButton
+                        type="primary"
+                        status="alt"
+                        htmlType="submit"
+                        data-attr="password-signup"
+                        fullWidth
+                        center
+                        loading={isSignupSubmitting || precheckResponseLoading}
+                        size="large"
+                    >
+                        Continue
+                    </LemonButton>
+                ) : (
+                    <SSOEnforcedLoginButton
+                        provider={precheckResponse.sso_enforcement}
+                        email={login.email}
+                        actionText="Continue"
+                        extraQueryParams={invite ? { invite_id: invite.id } : undefined}
+                    />
+                )}
+                {precheckResponse.saml_available && !precheckResponse.sso_enforcement && (
```

Comment:
And is this conditional correct? If we **don't** have `precheckResponse.sso_enforcement` we will show the `SSOEnforcedLoginButton` with `provider="saml"`?

Reply:
@raquelmsmith, it was correct, but I agree that the logic was hard to follow. Can you give it one final review?